backend/Application/Api/GraphQL/Network/NetworkQuery.cs

```diff
@@ -7,12 +7,33 @@ namespace Application.Api.GraphQL.Network;
 [ExtendObjectType(typeof(Query))]
 public class NetworkQuery
 {
+    private const int MaxSemverLength = 1024;
+
     [UsePaging]
-    public IEnumerable<NodeStatus> GetNodeStatuses([Service] NodeStatusSnapshot nodeSummarySnapshot)
+    public IEnumerable<NodeStatus> GetNodeStatuses(
+        [Service] NodeStatusSnapshot nodeSummarySnapshot,
+        NodeSortField sortField,
+        NodeSortDirection sortDirection)
     {
-        return nodeSummarySnapshot.NodeStatuses;
+        var statuses = nodeSummarySnapshot.NodeStatuses.AsQueryable();
+        statuses = sortField switch
+        {
+            NodeSortField.AveragePing => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.AveragePing) : statuses.OrderByDescending(s => s.AveragePing),
+            NodeSortField.BlocksReceivedCount => sortDirection == NodeSortDirection.ASC ? statuses.OrderBy(s => s.BlocksReceivedCount) : statuses.OrderByDescending(s => s.BlocksReceivedCount),
+            NodeSortField.ClientVersion => sortDirection == NodeSortDirection.ASC
+                ? statuses.OrderBy(s => Semver.SemVersion.Parse(s.ClientVersion, Semver.SemVersionStyles.Any, MaxSemverLength), Semver.SemVersion.SortOrderComparer)
+                : statuses.OrderByDescending(s => Semver.SemVersion.Parse(s.ClientVersion, Semver.SemVersionStyles.Any, MaxSemverLength), Semver.SemVersion.SortOrderComparer),
```

Comment:
At present there is no **guarantee** that the collector backend only returns valid SemVer for the client version field.

So in principle this is problematic. What will happen in that case? Will the exception be intercepted and the query will fail?

Reply:
The attached link says that is_compatible_version should be semver

Reply:
Also can you help us with what the format of the version can be? So that we can find an appropriate parser for the same

Reply:
At present there is no guarantee about the format. I would recommend the following logic

To compare versions v1, v2 do the following.
- attempt to parse v1, v2 as SemVer
- if successful compare them as SemVer
- if not then
- if v1 is valid SemVer then v1 < v2
- else both are not valid semVer, compare them as strings.

Reply:
@abizjak  - we have followed the steps that you have mentioned. and added tests
however we do think that the versions can only be a `SemVer`  since Semver type is being used at multiple places including the Handshake
https://github.com/Concordium/concordium-node/blob/0fc12c7b4392477afaa61d64595c947175f70803/concordium-node/src/network/mod.rs#L84

Is there something incorrect with that reasoning?
Will the nodes be able to talk to each other if a node publishes any other format for its version?

Reply:
You reasoning is correct assuming you would get the version from the node. But you do not get it from the node. You get the version from the "collector backend" which currently does not seem to do any validation on the client version.

That should also be fixed, but it is better that we are robust in ccdscan as well.